wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/gearsbot/subsystems/Claw.java

```diff
@@ -30,18 +33,19 @@ public void log() {
   }
 
   /** Set the claw motor to move in the open direction. */
-  public void open() {
-    m_motor.set(-1);
+  public Command open() {
+    return runOnce(() -> m_motor.set(-1))
+        .andThen(Commands.waitSeconds(1))
+        .andThen(m_motor::stopMotor)
+        .withName("open claw");
   }
 
   /** Set the claw motor to move in the close direction. */
-  public void close() {
-    m_motor.set(1);
-  }
-
-  /** Stops the claw motor from moving. */
-  public void stop() {
-    m_motor.set(0);
+  public Command close() {
+    return run(() -> m_motor.set(1))
+        .until(this::isGrabbing)
+        .andThen(m_motor::stopMotor)
```

Comment:
```suggestion
    return startEnd(() -> m_motor.set(1), m_motor::stopMotor)
        .until(this::isGrabbing)
```
Alternatively, if you think it's clearer:
```suggestion
    return runOnce(() -> m_motor.set(1))
        .until(Commands.waitUntil(this::isGrabbing))
        .andThen(m_motor::stopMotor)
```
Either of these will avoid repeatedly calling `m_motor.set(1)` every loop .

Reply:
The first is better because the motor will be stopped even if the command is interrupted.

Reply:
Good point about the interruption behavior! Some of the other command factories might have similar issues.

As I recall, repeatedly calling `set()` was discouraged because of increased bus traffic- Not necessarily an error, but preferable to avoid in examples.

Reply:
I'd expect vendordeps to avoid sending a frame if the value was identical, but they're closed source, so we have no idea.